service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -300,20 +394,19 @@ public Optional<EhrExtractStatus> updateEhrExtractStatusContinue(String conversa
     public void updateEhrExtractStatusAck(String conversationId, EhrReceivedAcknowledgement ack) {
 
         if (ack.getErrors() == null && !isEhrStatusWaitingForFinalAck(conversationId)) {
-            LOGGER.warn("Received unexpected acknowledgement of EHR Extract with conversation id=" + conversationId);
+            logger().warn("Received unexpected acknowledgement of EHR Extract with conversation id=" + conversationId);
             return;
         }
 
-        if (hasFinalAckBeenReceived(conversationId)) {
-            LOGGER.warn("Received an ACK message with a conversation_id=" + conversationId + " that is a duplicate");
+        if (hasAcknowledgementExceededEightDays(conversationId, ack.getReceived())
+            && hasEhrStatusReceivedAckWithUnexpectedConditionErrors(conversationId)) {
```

Comment:
Once you've renamed `hasEhrStatusReceivedAckWithUnexpectedConditionErrors` to something clearer, then I don't think you need the first check.